### PR TITLE
Deprecated the NumberToLocalizedStringTransformer constants

### LINKED ISSUE #13646: [Form] deprecate `NumberToLocalizedStringTransformer::ROUND_*` constants
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#36711
| PR author(s) | @nicolas-grekas
| Merged in    | 5.1-dev

### PR DESCRIPTION
**Pull request by javiereguiluz**
Fixes #13646.

### CODE REVIEW

File: reference/forms/types/options/rounding_mode.rst.inc [hunk 1]
```diff
@@ -1,32 +1,38 @@
 rounding_mode
 ~~~~~~~~~~~~~
 
-**type**: ``integer`` **default**: ``NumberToLocalizedStringTransformer::ROUND_HALF_UP``
+**type**: ``integer`` **default**: ``\NumberFormatter::ROUND_HALF_UP``
```
Comment (xabbuh) [inline reference/forms/types/options/rounding_mode.rst.inc:4]: `\NumberFormatter::ROUND_HALFUP`

File: reference/forms/types/options/rounding_mode.rst.inc [hunk 2]
```diff
@@ -1,32 +1,38 @@
...
-* ``NumberToLocalizedStringTransformer::ROUND_UP`` Round away from zero. It
+* ``\NumberFormatter::ROUND_UP`` Round away from zero. It
   rounds ``1.4`` to ``2`` and ``-1.4`` to ``-2``.
 
-* ``NumberToLocalizedStringTransformer::ROUND_CEILING`` Round towards positive
+* ``\NumberFormatter::ROUND_CEILING`` Round towards positive
   infinity. It rounds ``1.4`` to ``2`` and ``-1.4`` to ``-1``.
 
-* ``NumberToLocalizedStringTransformer::ROUND_HALF_DOWN`` Round towards the
+* ``\NumberFormatter::ROUND_HALF_DOWN`` Round towards the
```
Comment (xabbuh) [inline reference/forms/types/options/rounding_mode.rst.inc:22]: `\NumberFormatter::ROUND_HALFDOWN`

File: reference/forms/types/options/rounding_mode.rst.inc [hunk 3]
```diff
@@ -1,32 +1,38 @@
...
   "nearest neighbor". If both neighbors are equidistant, round down. It rounds
   ``2.5`` and ``1.6`` to ``2``, ``1.5`` and ``1.4`` to ``1``.
 
-* ``NumberToLocalizedStringTransformer::ROUND_HALF_EVEN`` Round towards the
+* ``\NumberFormatter::ROUND_HALF_EVEN`` Round towards the
```
Comment (xabbuh) [inline reference/forms/types/options/rounding_mode.rst.inc:26]: `\NumberFormatter::ROUND_HALFEVEN`

File: reference/forms/types/options/rounding_mode.rst.inc [hunk 4]
```diff
@@ -1,32 +1,38 @@
...
   "nearest neighbor". If both neighbors are equidistant, round towards the even
   neighbor. It rounds ``2.5``, ``1.6`` and ``1.5`` to ``2`` and ``1.4`` to ``1``.
 
-* ``NumberToLocalizedStringTransformer::ROUND_HALF_UP`` Round towards the
+* ``\NumberFormatter::ROUND_HALF_UP`` Round towards the
```
Comment (xabbuh) [inline reference/forms/types/options/rounding_mode.rst.inc:30]: `\NumberFormatter::ROUND_HALFUP`

File: reference/forms/types/options/rounding_mode.rst.inc [hunk 5]
```diff
@@ -1,32 +1,38 @@
...
   "nearest neighbor". If both neighbors are equidistant, round up. It rounds
   ``2.5`` to ``3``, ``1.6`` and ``1.5`` to ``2`` and ``1.4`` to ``1``.
+
+.. versionadded:: 5.1
+
+    The usage of ``\NumberFormatter`` constants was introduced in Symfony 5.1.
```
Comment (xabbuh) [inline reference/forms/types/options/rounding_mode.rst.inc:36]: I wouldn't word it this way. You could already use the constants from the `NumberFormatter` class before as the constants inside `NumberToLocalizedStringTransformer` were just aliases for them.

**Review by xabbuh — CHANGES_REQUESTED** [review]

**Comment by javiereguiluz** [comment]
Christian, thanks for the fixes! I've also reworded the last paragraph.

**Review by xabbuh — APPROVED** [review]

File: reference/forms/types/options/rounding_mode.rst.inc (hunk 1 above)
Comment (xabbuh) [inline reference/forms/types/options/rounding_mode.rst.inc:4]: one change is missing here: `\NumberFormatter::ROUND_HALFUP`

File: reference/forms/types/options/rounding_mode.rst.inc [hunk 6]
```diff
@@ -1,32 +1,38 @@
...
 
-**type**: ``integer`` **default**: ``NumberToLocalizedStringTransformer::ROUND_HALF_UP``
+**type**: ``integer`` **default**: ``\NumberFormatter::ROUND_HALFUP``
 
-If a submitted number needs to be rounded (based on the `scale`_
-option), you have several configurable options for that rounding. Each
-option is a constant on the :class:`Symfony\\Component\\Form\\Extension\\Core\\DataTransformer\\NumberToLocalizedStringTransformer`:
+If a submitted number needs to be rounded (based on the `scale`_ option), you
+have several configurable options for that rounding. Each option is a constant
+on the ``\NumberFormatter`` class:
```
Comment (wouterj) [inline reference/forms/types/options/rounding_mode.rst.inc:8]: What about doing ``:phpclass:`NumberFormatter` `` here?

Comment (javiereguiluz) [reply reference/forms/types/options/rounding_mode.rst.inc:8]: Yes! Done in af24efab9a3b48323f0cece7eb8944d6d8687054

**Pull request merged**